os/arch/arm/src/armv7-a/arm_gicv2.c

```diff
@@ -508,6 +508,43 @@ int up_prioritize_irq(int irq, int priority)
 	return -EINVAL;
 }
 
+/****************************************************************************
+ * Name: up_affinity_irq
+ *
+ * Description:
+ *   set the affinity of a given irq (SPI only) to given cpu.
+ *   Currenlty, it doesnot support assigning multiple cpus as affinity.
+ *   It can assign only one cpu per SPI.
+ *
+ * Input Parameters:
+ *   irq: irq id
+ *   cpu: cpu id
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void up_affinity_irq(int irq, int cpu)
+{
+	DEBUGASSERT(irq >= GIC_IRQ_SPI && irq < NR_IRQS && cpu >= 0 && cpu < CONFIG_SMP_NCPUS);
```

Comment:
Let us split this into 2 lines for make debugging simpler. One line for irq and other for cpu.